File: template_backend/src/main/java/com/benny1611/template/dto/ApiError.java
package com.benny1611.template.dto;

import lombok.Data;

import java.time.Instant;

@Data
public class ApiError {
    private final int status;
    private final String error;
    private final String message;
    private final String path;
    private final Instant timestamp = Instant.now();
}

File: template_frontend/easyevent/src/models/dto/ApiError.ts
export class ApiError {
    status: number;
    error: string;
    message: string;
    path: string;
    timestamp: string;

    constructor(status: number, error: string, message: string, path: string, timestamp: string) {
        this.status = status;
        this.error = error;
        this.message = message;
        this.path = path;
        this.timestamp = timestamp;
    }
}

export default ApiError;